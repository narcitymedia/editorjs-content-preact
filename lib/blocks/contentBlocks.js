export var BlockType;
(function (BlockType) {
    BlockType["Header"] = "header";
    BlockType["Paragraph"] = "paragraph";
    BlockType["List"] = "list";
    BlockType["RawHTML"] = "raw";
    BlockType["Delimiter"] = "delimiter";
})(BlockType || (BlockType = {}));
export var ListStyle;
(function (ListStyle) {
    ListStyle["Ordered"] = "ordered";
    ListStyle["Unordered"] = "unordered";
})(ListStyle || (ListStyle = {}));
