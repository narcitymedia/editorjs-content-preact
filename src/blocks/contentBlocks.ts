export enum BlockType {
    Header = "header",
    Paragraph = "paragraph",
    List = "list",
    RawHTML= "raw",
    Delimiter = "delimiter"
}

export interface ContentBlock {
    /**
     * Represents the type of block
     */
    type: string,

    /**
     * Data associated with the current block
     */
    data: unknown

    /**
     * Uniquely identifies the block within the content
     */
    contentUniqueId: string
}

export interface HeaderBlock extends ContentBlock {
    
}

export interface ParagraphBlock extends ContentBlock {
    data: {
        /**
         * Text contained in the current paragraph. May contain markup.
         */
        text: string
    }
}
