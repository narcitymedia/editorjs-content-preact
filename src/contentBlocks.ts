export interface ContentBlock {
    /**
     * Represents the type of block
     */
    type: string,

    /**
     * Data associated with the current block
     */
    data: any
}

export interface Header extends ContentBlock {
    
}
