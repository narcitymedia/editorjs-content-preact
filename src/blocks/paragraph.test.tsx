import React from 'react';
import * as renderer from 'react-test-renderer';

import Paragraph from './paragraph';
import { ParagraphBlock, BlockType } from './contentBlocks';

describe('Paragraph', () => {
  it('should render the Paragraph with no markup', () => {
    const args: ParagraphBlock = {
      contentUniqueId: 'XXX-XXX',
      type: BlockType.RawHTML,
      data: {
        text: 'No markup paragraph',
      },
    };

    const tree = renderer.create(<Paragraph {...args} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render the Paragraph with markup', () => {
    const args: ParagraphBlock = {
      contentUniqueId: 'XXX-XXX',
      type: BlockType.RawHTML,
      data: {
        text: "No <b>markup</b> paragraph <a href='https://www.google.ca'>Google Link</a>",
      },
    };

    const tree = renderer.create(<Paragraph {...args} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
