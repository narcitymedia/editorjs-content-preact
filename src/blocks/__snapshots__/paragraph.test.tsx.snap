// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should render the Paragraph with markup 1`] = `
<p
  dangerouslySetInnerHTML={
    Object {
      "__html": "No <b>markup</b> paragraph <a href='https://www.google.ca'>Google Link</a>",
    }
  }
  id="XXX-XXX"
/>
`;

exports[`Paragraph should render the Paragraph with no markup 1`] = `
<p
  dangerouslySetInnerHTML={
    Object {
      "__html": "No markup paragraph",
    }
  }
  id="XXX-XXX"
/>
`;
